Test/Classes/TestTextUtils.cs

```diff
@@ -409,5 +409,27 @@ public void StripDateFromNameIgnoreDifferent()
             Assert.Equal("22/02/2009 Test", TextUtils.StripDateFromName("22/02/2009 Test", date));
             Assert.Equal("Test 22/02/2009 Test", TextUtils.StripDateFromName("Test 22/02/2009 Test", date));
         }
+
+        /// <summary>
+        /// Test that the BuildEpisodeSmartName function correctly normalises
+        /// episode names
+        /// </summary>
+        [Fact]
+        public void BuildEpisodeSmartName()
+        {
+            DateTime date = new DateTime(2017, 12, 05);
+
+            Assert.Equal("Programme 1: Episode 1", TextUtils.BuildEpisodeSmartName("Programme 1", "Episode 1", date));
+            Assert.Equal("Programme 2: Episode 1", TextUtils.BuildEpisodeSmartName("Programme 2", "Programme 2: Episode 1", date));
+            Assert.Equal("Programme 3: Episode 1", TextUtils.BuildEpisodeSmartName("Programme 3", "Programme 3 - Episode 1 - 05/12/2017", date));
+            Assert.Equal("Programme 4: Episode 1", TextUtils.BuildEpisodeSmartName("Programme 4", "Episode 1: 05/12/2017", date));
+            Assert.Equal("Programme 5", TextUtils.BuildEpisodeSmartName("Programme 5", "05/12/2017", date));
+            Assert.Equal("Programme 6", TextUtils.BuildEpisodeSmartName("Programme 6", "Programme 6", date));
+            Assert.Equal("Programme 7", TextUtils.BuildEpisodeSmartName("Programme 7", "Programme 7 05/12/2017", date));
+            Assert.Equal("Programme 8", TextUtils.BuildEpisodeSmartName("Programme 8", "Programme 8, 05/12/2017", date));
```

Comment:
Was there a reason that you changed these tests from my originals?

    Assert.Equal("Programme 7", TextUtils.BuildEpisodeSmartName("Programme 7", "Programme 7, 05/12/2017", date));
    Assert.Equal("Programme - 8: Episode - 1", TextUtils.BuildEpisodeSmartName("Programme - 8", "Episode - 1", date));

Reply:
Yes, only in as much as the `BuildEpisodeSmartName` was failing to match the data value on the test that had function test `Assert.Equal("Programme 7", TextUtils.BuildEpisodeSmartName("Programme 7", "Programme 7, 05/12/2017", date));`. 

Thus split this test out over two tests (Programme 7 and Programme 8, and renamed your test Programme 8 to Programme 9) as a means to work through the handling of the comma delimiter between the programme name and date `Programme 7, 05/12/2017`. 

With that resolved should have restored your test case back (happy to do so).